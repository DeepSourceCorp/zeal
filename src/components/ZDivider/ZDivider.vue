<template>
  <div
    :class="['z-divider', `z-divider--${direction}`]"
    class="relative block h-px w-full my-6 mx-0 bg-ink-100"
  >
    <div class="z-divider--content" v-if="direction !== 'vertical'"><slot></slot></div>
  </div>
</template>

<script lang="ts">
export default {
  name: 'ZDivider',
  props: {
    direction: {
      default: 'horizontal',
      type: String
    }
  }
}
</script>

<style lang="css" scoped>
.z-divider--vertical {
  @apply inline-block w-px h-4 my-0 mx-2 align-middle;
}
</style>
