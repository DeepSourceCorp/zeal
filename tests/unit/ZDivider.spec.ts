import ZDivider from '../../src/components/ZDivider'
import Vue from 'vue'

import { mount, Wrapper } from '@vue/test-utils'

describe('ZDivider', () => {
  let mountFn: (options?: Record<string, unknown>) => Wrapper<Vue>
  let localVue: typeof Vue

  beforeEach(() => {
    mountFn = (options = {}) => {
      return mount(ZDivider, {
        localVue,
        ...options
      })
    }
  })

  it('renders the component', () => {
    const wrapper = mountFn()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a vertical divider', () => {
    const wrapper = mountFn({
      propsData: {
        direction: 'vertical'
      }
    })

    expect(wrapper.classes()).toContain('h-4')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
