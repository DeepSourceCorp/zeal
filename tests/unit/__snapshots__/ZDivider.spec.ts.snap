// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZDivider renders a vertical divider 1`] = `
<div class="relative bg-ink-200 z-divider inline-block w-px h-4 my-0 mx-2 align-middle">
  <!---->
</div>
`;

exports[`ZDivider renders the component 1`] = `
<div class="relative bg-ink-200 z-divider block h-px w-full my-6 mx-0">
  <div class="z-divider--content"></div>
</div>
`;
